app/src/main/java/com/randomappsinc/studentpicker/presentation/PresentationAdapter.kt

```diff
@@ -0,0 +1,92 @@
+package com.randomappsinc.studentpicker.presentation
+
+import android.graphics.Color
+import android.text.TextUtils
+import android.util.TypedValue
+import android.view.Gravity
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.LinearLayout
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import com.randomappsinc.studentpicker.databinding.ChooseNameDialogCellBinding
+import com.randomappsinc.studentpicker.models.NameDO
+import com.randomappsinc.studentpicker.utils.NameUtils
+import com.squareup.picasso.Picasso
+
+class PresentationAdapter : RecyclerView.Adapter<PresentationAdapter.ViewHolder>() {
+
+    var chosenNames = ArrayList<NameDO>()
+    var showAsList = false
+
+    fun setChosenNames(chosenNames: List<NameDO>, showAsList: Boolean) {
+        this.showAsList = showAsList
+        this.chosenNames.clear()
+        this.chosenNames.addAll(chosenNames)
+        notifyDataSetChanged()
+    }
+
+    var textSize = 27f
+        set(value) {
+            field = value
+            notifyDataSetChanged()
+        }
+
+    var textColor = Color.GRAY
+        set(value) {
+            field = value
+            notifyDataSetChanged()
+        }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+        return ViewHolder(
+                ChooseNameDialogCellBinding.inflate(
+                        LayoutInflater.from(parent.context), parent, false))
+    }
+
+    override fun getItemCount(): Int {
+        return chosenNames.size
+    }
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        holder.loadName(position)
+    }
+
+    inner class ViewHolder(itemBinding: ChooseNameDialogCellBinding) : RecyclerView.ViewHolder(itemBinding.root) {
+
+        private val parent: LinearLayout = itemBinding.root
+        private val nameNumber: TextView = itemBinding.personNumber
+        private val personImageView: ImageView = itemBinding.personImage
+        private val name: TextView = itemBinding.personName
+
+        fun loadName(position: Int) {
+            val nameDO: NameDO = chosenNames[position]
+            if (showAsList) {
```

Comment:
Show as list can't change while the user is on the screen, so technically you can move this into onCreateViewHolder where the view is inflated so you don't need to do it again on this view being recycled and rebound.